### PR TITLE
new custom type and key map

### PR DESCRIPTION
**Pull request by bojiguard**
add custom type
add new keymap
     for via
    for vial

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [x] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/fungo/frankenstein/config.h [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+/*************************************/
+/**       public parameter       **/
+/*************************************/
+
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
```
Comment (waffle87) [inline keyboards/fungo/frankenstein/config.h:33]:
```suggestion
```

File: keyboards/fungo/frankenstein/config.h [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+//caps word, left shift + right shift => capslock
+#define BOTH_SHIFTS_TURNS_ON_CAPS_WORD
+#define CAPS_WORD_IDLE_TIMEOUT 500   //default 5000msec
+
+//tap or holding
+#define TAPPING_TERM 200
+
```
Comment (waffle87) [inline keyboards/fungo/frankenstein/config.h:41]:
```suggestion
```

File: keyboards/fungo/frankenstein/config.h [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+#define PIMORONI_TRACKBALL_ADDRESS 	0x0A
+#define PIMORONI_TRACKBALL_TIMEOUT 	100
+#define PIMORONI_TRACKBALL_SCALE 	5
+#define PIMORONI_TRACKBALL_DEBOUNCE_CYCLES 	20
+#define PIMORONI_TRACKBALL_ERROR_COUNT 	10
+
+
+//
```
Comment (waffle87) [inline keyboards/fungo/frankenstein/config.h:55]:
```suggestion
```

File: keyboards/fungo/frankenstein/config.h [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+#define ENCODERS_PAD_A { D1 }
+#define ENCODERS_PAD_B { D0 }
+#define ENCODER_RESOLUTIONS { 4 }
+#define ENCODERS_PAD_A_RIGHT { }
+#define ENCODERS_PAD_B_RIGHT { }
+#define ENCODER_RESOLUTIONS_RIGHT { }
+#define ENCODER_DIRECTION_FLIP
```
Comment (waffle87) [inline keyboards/fungo/frankenstein/config.h:62]:
```suggestion
```

File: keyboards/fungo/frankenstein/info.json [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+    "pid": "0x1234",
+    "device_version": "0.0.1"
+  },
+  "matrix_pins": {
+    "cols": ["F4", "F5", "F6", "F7", "B1", "B3", "B2"],
+    "rows": ["D4", "C6", "D7", "E6", "B4", "B5"],
+    "io_delay": 15
+  },
+  "diode_direction": "COL2ROW",
+  "split": {
```
Comment (waffle87) [inline keyboards/fungo/frankenstein/info.json:17]:
```suggestion
  "split": {
    "enabled": true,
```

File: keyboards/fungo/frankenstein/info.json [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+      "right": {
+        "cols": ["F7", "F6", "F5", "F4", "B5", "B4", "E6"],
+        "rows": ["B1", "B3", "D4", "C6", "D7", "D2"]
+      }
+    }
+  },
+  "features": {
+  },
+  "qmk": {
+    "tap_keycode_delay": 50
```
Comment (waffle87) [inline keyboards/fungo/frankenstein/info.json:29]:
```suggestion
    "tap_keycode_delay": 50,
    "locking": {
      "enabled": true,
      "resync": true
    }
```

File: keyboards/fungo/frankenstein/post_config.h
Comment (waffle87) [inline keyboards/fungo/frankenstein/post_config.h:1]: This file should simply be removed.

File: keyboards/fungo/frankenstein/config.h [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+#define CAPS_WORD_IDLE_TIMEOUT 500   //default 5000msec
+
+//tap or holding
+#define TAPPING_TERM 200
+
+#define POINTING_DEVICE_RIGHT
+#define SPLIT_POINTING_ENABLE
+#define POINTING_DEVICE_ROTATION_90
+
+
```
Comment (waffle87) [inline keyboards/fungo/frankenstein/config.h:46]:
```suggestion
#define MK_3_SPEED
#define MOUSEKEY_INTERVAL 3
#define MOUSEKEY_INITIAL_SPEED 20
```

File: keyboards/fungo/frankenstein/info.json [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+  },
+  "processor": "atmega32u4",
+  "bootloader": "atmel-dfu",
```
Comment (waffle87) [inline keyboards/fungo/frankenstein/info.json:32]:
```suggestion
  "development_board": "elite_c",
```

File: keyboards/fungo/frankenstein/rules.mk [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+ENCODER_ENABLE = yes        # encoder
+ENCODER_MAP_ENABLE = yes    # ENCODER MAP
+
+# caps word, auto return prev capslock,  
+# check option {COMMAND_ENABLE = no}
+CAPS_WORD_ENABLE = yes
+
+POINTING_DEVICE_ENABLE = yes
+# POINTING_DEVICE_DRIVER = analog_joystick
+POINTING_DEVICE_DRIVER = pimoroni_trackball
```
Comment (waffle87) [inline keyboards/fungo/frankenstein/rules.mk:31]:
```suggestion
POINTING_DEVICE_DRIVER = pimoroni_trackball
```

File: keyboards/fungo/frankenstein/info.json [hunk 10]
```diff
@@ -0,0 +1,130 @@
...
+  "split": {
+    "soft_serial_pin": "D3",
+    "matrix_pins": {
+      "right": {
+        "cols": ["F7", "F6", "F5", "F4", "B5", "B4", "E6"],
+        "rows": ["B1", "B3", "D4", "C6", "D7", "D2"]
+      }
+    }
+  },
+  "features": {
```
Comment (waffle87) [inline keyboards/fungo/frankenstein/info.json:26]:
```suggestion
  "features": {
    "bootmagic": true,
    "extrakey": true,
    "mousekey": true,
    "encoder": true,
    "pointing_device": true,
    "key_lock": true,
    "caps_word": true
```

File: keyboards/fungo/frankenstein/info.json [hunk 11]
```diff
@@ -0,0 +1,130 @@
...
+  },
```
Comment (waffle87) [inline keyboards/fungo/frankenstein/info.json:27]:
```suggestion
  },
  "caps_word": {
    "both_shifts_turn_on": true,
    "idle_timeout": 500
  },
```

File: keyboards/fungo/frankenstein/config.h [hunk 12]
```diff
@@ -0,0 +1,62 @@
...
+/* Mechanical locking support. Use KC_LCAP, KC_LNUM or KC_LSCR instead in keymap */
+#define LOCKING_SUPPORT_ENABLE
+
+/* Locking resynchronize hack */
+#define LOCKING_RESYNC_ENABLE
+
+//caps word, left shift + right shift => capslock
+#define BOTH_SHIFTS_TURNS_ON_CAPS_WORD
+#define CAPS_WORD_IDLE_TIMEOUT 500   //default 5000msec
+
```
Comment (waffle87) [inline keyboards/fungo/frankenstein/config.h:38]:
```suggestion
```

File: keyboards/fungo/keymaps/frank/keymap.c
Comment (waffle87) [inline keyboards/fungo/keymaps/frank/keymap.c:1]: This directory must be removed, as custom user keymaps are no longer accepted.

File: keyboards/fungo/keymaps/via_frank/fungo_frankenstein.json
Comment (waffle87) [inline keyboards/fungo/keymaps/via_frank/fungo_frankenstein.json:1]: This directory must be removed, as custom user keymaps are no longer accepted.

File: keyboards/fungo/keymaps/vial/config.h
Comment (waffle87) [inline keyboards/fungo/keymaps/vial/config.h:1]: This directory must be removed, as VIAL keymaps are **not** accepted.

**Review by waffle87 — CHANGES_REQUESTED** [review]
This keyboard is missing a `default` keymap, which is required. Please thoroughly review the PR checklist, as many things have been ignored.

File: keyboards/fungo/keymaps/via_frank/fungo_frankenstein.json
Comment (bojiguard) [reply keyboards/fungo/keymaps/via_frank/fungo_frankenstein.json:1]: what is mean?
Is the folder name via_frank not acceptable?
Then, how can I change it?

Comment (waffle87) [reply keyboards/fungo/keymaps/via_frank/fungo_frankenstein.json:1]: The folder name can be renamed to just `via`. All the commented out code should also be removed.

File: keyboards/fungo/frankenstein/config.h [hunk 13]
```diff
@@ -0,0 +1,28 @@
...
+/* select keyboard  master board  - I2C or Serial communication master    */
+#define MASTER_RIGHT
+
+#define POINTING_DEVICE_RIGHT
+#define SPLIT_POINTING_ENABLE
+#define POINTING_DEVICE_ROTATION_90
+
+#define ENCODERS_PAD_A { D1 }
+#define ENCODERS_PAD_B { D0 }
```
Comment (waffle87) [inline keyboards/fungo/frankenstein/config.h:28]:
```suggestion
```

File: keyboards/fungo/frankenstein/info.json [hunk 14]
```diff
@@ -0,0 +1,135 @@
...
+  "split": {
+    "enabled": true,
+    "soft_serial_pin": "D3",
+    "matrix_pins": {
+      "right": {
+        "cols": ["F7", "F6", "F5", "F4", "B5", "B4", "E6"],
+        "rows": ["B1", "B3", "D4", "C6", "D7", "D2"]
+      }
+    }
+  },
```
Comment (waffle87) [inline keyboards/fungo/frankenstein/info.json:26]:
```suggestion
  },
  "encoder": {
    "rotary": [
      { "pin_a": "D1", "pin_b": "D0" }
    ]
  },
```

File: keyboards/fungo/keymaps/via_frank/fungo_frankenstein.json
Comment (waffle87) [inline keyboards/fungo/keymaps/via_frank/fungo_frankenstein.json:1]: This file should be removed, it does not belong in the qmk_firmware repository.

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/fungo/keymaps/via_frank/fungo_frankenstein.json
Comment (bojiguard) [reply keyboards/fungo/keymaps/via_frank/fungo_frankenstein.json:1]: Do I have to use the folder name as via to use via-keymap?
What if via-keymap is different for each keyboard version?
Why is the via-key map of this allowed as "mb_via"?
---> crkbd/keymaps/mb_via

**Comment by tzarc** [comment]
At this stage the PR is nowhere near mergeable. Please read the PR checklist, make the required changes, then re-read it again to be sure.

In its current state this will be closed by stalebot.

**Pull request closed without merging**